### ISSUE TITLE
Index file for var prefix variables.data.cont-8.best does not exist

### ISSUE THREAD

**Issue by mlappe** [ufal/neuralmonkey#541]
Hello :)
I trained a translation model and now I am trying to use neuralmonkey-run to get translations for my test set.
unfortunately i can not load my checkpoints, I get the following message instead:

> Index file for var prefix variables.data.cont-8.best does not exist

that's the code snippet where the message gets printed
```
for vfile in variable_files:
    if not os.path.exists("{}.index".format(vfile)):
       log("Index file for var prefix {} does not exist".format(vfile),
          color="red")
       exit(1)
```
so nm looks for a file named:

> variables.data.cont-8.best.index

which I don't have, but I do have a file named

> variables.data.cont-8.index   (no best inbetween)

I run some of your tests and I never get a .best.index file, just a .index ,.best  and .meta
It would be great if you could help me
thanks in advance

my ini
```
>[main]
name="Spielwiese"
tf_manager=<tf_manager>
output="tests/outputs/spielwiese"
batch_size=16
epochs=50
train_dataset=<train_data>
val_dataset=<val_data>
trainer=<trainer>
runners=[<runner>]
evaluation=[("target", <bleu>)]
overwrite_output_dir=True



[tf_manager]
class=tf_manager.TensorFlowManager
num_sessions=1
num_threads=1

[bleu]
class=evaluators.bleu.BLEUEvaluator


[train_data]
class=dataset.load_dataset_from_files
s_source="tests/data/Spielwiesedata/train/source"
s_target="tests/data/Spielwiesedata/train/target"

[val_data]
class=dataset.load_dataset_from_files
s_source="tests/data/Spielwiesedata/dev/source"
s_target="tests/data/Spielwiesedata/dev/target"


[encoder]
class=encoders.recurrent.SentenceEncoder
name="sentence_encoder"
rnn_size=50
max_input_len=10
embedding_size=10
dropout_keep_prob=0.5
attention_type=decoding_function.Attention
data_id="source"
vocabulary=<encoder_vocabulary>

[decoder]
class=decoders.decoder.Decoder
name="decoder"
max_output_len=10
encoders=[<encoder>]
rnn_size=10
embedding_size=10
use_attention=True
dropout_keep_prob=0.5
data_id="target"
vocabulary=<decoder_vocabulary>

[encoder_vocabulary]
class=vocabulary.from_dataset
datasets=[<train_data>]
series_ids=["source"]
max_size=60
save_file="tests/outputs/spielwiese/encoder_vocab"
overwrite=True

[decoder_vocabulary]
class=vocabulary.from_dataset
datasets=[<train_data>]
series_ids=["target"]
max_size=70
save_file="tests/outputs/spielwiese/decoder_vocab"
overwrite=True

[trainer]
class=trainers.cross_entropy_trainer.CrossEntropyTrainer
decoders=[<decoder>]
l2_weight=1.0e-8

[runner]
class=runners.runner.GreedyRunner
output_series="target"
decoder=<decoder>
```

my test data ini
```
[main]
test_datasets=[<eval_data>]

[eval_data]
class=dataset.load_dataset_from_files
s_source="tests/data/Spielwiesedata/test/source"
s_greedy_edits_out="tests/outputs/spielwiese/outputtext"
```

**Comment by jindrahelcl** [comment]
Hi, sorry, this bit is not yet well documented, so I will try and explain it here:

There are two methods of setting the variables used for inference:

1. Default - if you don't set the variable location in the configuration, it search the output directory for a couple of files with a common prefix `variables.data`. As I can see now, there is a bug, however, if you try to use this when storing multiple experiments in the same directory (i.e. when the variable have those `.cont-*` suffixes). So doing this is not possible in the current version.

2. Explicit - you can specify which variable files to use directly in the test data ini, in the `[main]` section.
For this, use the  `variables` option, whose value is a list of strings, in which every string is a path to the variable files. Since there are three variable files generated by tensorflow, the tensorflow saver accepts only the common prefix (which may not be an existing file!), in our case, the `variable.data.cont-8`

So, change your test data file accordingly:
```
[main]
test_datasets=[<eval_data>]
variables=["tests/outputs/spielwiese/variables.data.cont-8"]

[eval_data]
class=dataset.load_dataset_from_files
s_source="tests/data/Spielwiesedata/test/source"
s_greedy_edits_out="tests/outputs/spielwiese/outputtext"
```

Note: you can try giving a number of files to perform ensembling, but it does not work well in the current master. If you decide to try ensembles, checkout the `ensembles` branch first.


Take-home messages: 
1. don't use default var file resolution when sharing one experiment dir across multiple runs
2. variable files are not looked up by an actual filename, but rather just by a common prefix (which must match those three files (i.e. `*.index`, `*.meta`, and `*.data-00000-of-00001`))


Thanks for your interest in the toolkit.. :-)

**Comment by mlappe** [comment]
thanks for your fast answer :+1:
